### PR TITLE
Question about requests

### PR DESCRIPTION
**Pull request by teelu**
Issue with private pub
Some fixes are from previous PR's.  I just went on ahead

### CODE REVIEW

File: app/controllers/requests_controller.rb [hunk 1]
```diff
@@ -20,6 +20,16 @@ def create
...
+      PrivatePub.publish_to "/valet/new", :request => {id: @request.id,
+                                                name: "#{@request.user.first_name} #{@request.user.last_name}",
+                                                picture: @request.user.profile_picture,
+                                                transmission: @request.user.is_manual,
+                                                phone: @request.user.phone_number,
+                                                latitude: @request.source_location.latitude,
+                                                longitude: @request.source_location.longitude,
+                                                location: @request.source_location.address,
+                                                type: "pick_up"
+                                              }
```
Comment (teelu) [inline app/controllers/requests_controller.rb]: How can I render this in jbuilder?   I tried to use render but it spits out an error.  I looked at docs and it told me to do it through a j render on a erb file through a view

File: app/controllers/requests_controller.rb [hunk 2]
```diff
@@ -20,6 +20,18 @@ def create
...
+        id:             @request.id,
+        name:           "#{@request.user.first_name} #{@request.user.last_name}",
+        picture:        @request.user.profile_picture,
+        transmission:   @request.user.is_manual,
+        phone:          @request.user.phone_number,
+        latitude:       @request.source_location.latitude,
+        longitude:      @request.source_location.longitude,
+        location:       @request.source_location.address,
+        type:           "pick_up"
+      }
```
Comment (teelu) [inline app/controllers/requests_controller.rb]: Is there a better way to display this information?  I tried to render it in a jbuilder but it just gave me an error and the docs for private pub are not very clear for what to do.  Maybe I could publish_to inside the jbuilder file?

**Comment by f3r** [comment]
@teelu ping me up when you update this

File: app/controllers/requests_controller.rb (hunk 2 above)
Comment (h8rry) [inline app/controllers/requests_controller.rb]: Don't think there's jbuilder for it. Jbuilder is for views.

File: Gemfile [hunk 3]
```diff
@@ -15,7 +15,7 @@ gem 'dotenv-rails'
 gem 'rack-cors', :require => 'rack/cors'
 gem 'aasm'
 gem 'geocoder'
-gem 'private_pub'
+# gem 'private_pub'
 gem 'private_pub_no_sig'
```
Comment (h8rry) [inline Gemfile:19]: I thought you said the original one works.

File: app/views/requests/index_drop_off.jbuilder [hunk 4]
```diff
@@ -0,0 +1,25 @@
+json.array! @requests do |request|
+  json.id       request.id
+  json.user do
+    json.id               request.user.id
+    json.name             "#{request.user.first_name} #{request.user.last_name}"
+    json.profile_picture  request.user.profile_picture
+    if request.user.is_manual
```
Comment (h8rry) [inline app/views/requests/index_drop_off.jbuilder:7]: I am still against having the same information... `is_manual` and `transmission` are the same thing

**Comment by h8rry** [comment]
@teelu The rest looks fine. Please review + merge. Then, rebase #16  and ping me

**Comment by teelu** [comment]
@ggharry  please review the commits i made,  I got the ordering wrong on car_drop_off and fixed it as well as changing some data that is being sent through private pub and what is being rendered

**Pull request merged**